pkg/trace/stats/sublayers.go

```diff
@@ -10,7 +10,16 @@ import (
 	"sort"
 
 	"github.com/DataDog/datadog-agent/pkg/trace/pb"
-	"github.com/DataDog/datadog-agent/pkg/trace/traceutil"
+)
+
+const (
+	// defaultSublayersCalculatorMaxSpans is the maximum trace size in spans the calculator can process.
+	// if a bigger trace comes, the calculator will re-allocate bigger arrays.
+	defaultSublayersCalculatorMaxSpans = 10000
```

Comment:
```go
// defaultSublayersCalculatorMaxSpans is the maximum trace size in spans the calculator can process.
```
This sentence tells me that the sublayer calculator can not process a trace larger than 10000 spans.
```go
// if a bigger trace comes, the calculator will re-allocate bigger arrays.
```
The next sentence contradicts the previous one saying that it actually can process it, but will re-allocate.

This is a bit confusing and we can maybe improve it by also removing the word "max":
```go
// defaultCalculatorSpanCapacity specifies the maximum trace size in spans that the calculator
// can process without re-allocating.
defaultCalculatorSpanCapacity = 10000
```